bookkeeper-common/src/main/java/org/apache/bookkeeper/common/util/SingleThreadExecutor.java

```diff
@@ -225,6 +246,26 @@ public void execute(Runnable r) {
         }
     }
 
+    private void monitorTimeoutTasks() {
+        for (Runnable task : queue) {
```

Comment:
The `queue` is meant to be used by a single consumer, so we cannot iterate over it from a different thread.